File: Placements.DataAccess/placement/Models/Trainingcourse.cs
using System;
using System.Collections.Generic;

namespace Placements.DataAccess.placement.Models;

public partial class Trainingcourse
{
    public long Id { get; set; }

    public string? Name { get; set; }

    public string? Description { get; set; }

    public DateTime? ValidFrom { get; set; }

    public DateTime? ValidTill { get; set; }

    public ulong IsDeleted { get; set; }

    public virtual ICollection<Trainerschedule> Trainerschedules { get; set; } = new List<Trainerschedule>();

    public virtual ICollection<Trainingmodule> Trainingmodules { get; set; } = new List<Trainingmodule>();
}

File: web-app/src/app/services/types/Trainingcourse.ts
import { Trainerschedule } from "./Trainerschedule";
import { Trainingmodule } from "./Trainingmodule";

export type Trainingcourse = {
    Id: number;
    Name?: string;
    Description?: string;
    ValidFrom?: Date;
    ValidTill?: Date;
    IsDeleted: boolean;
    Trainerschedules?: Trainerschedule[];
    Trainingmodules?: Trainingmodule[];
};
